packages/lib/server/repository/selectedCalendar.ts

```diff
@@ -15,4 +17,79 @@ export class SelectedCalendarRepository {
       },
     });
   }
+  static async upsert(data: Prisma.SelectedCalendarUncheckedCreateInput) {
+    return await prisma.selectedCalendar.upsert({
+      where: {
+        userId_integration_externalId: {
+          userId: data.userId,
+          integration: data.integration,
+          externalId: data.externalId,
+        },
+      },
+      create: { ...data },
+      update: { ...data },
+    });
+  }
+  /** Retrieve calendars that need to be watched */
+  static async getNextBatchToWatch(limit = 100) {
+    // Get selected calendars from users that belong to a team that has calendar cache enabled
+    const oneDayInMS = 24 * 60 * 60 * 1000;
+    const tomorrowTimestamp = String(new Date().getTime() + oneDayInMS);
+    const nextBatch = await prisma.selectedCalendar.findMany({
+      take: limit,
+      where: {
+        user: {
+          teams: {
+            some: {
+              team: {
+                features: {
+                  some: {
+                    featureId: "calendar-cache",
+                  },
+                },
+              },
+            },
+          },
+        },
+        // RN we only support google calendar subscriptions for now
+        integration: "google_calendar",
+        AND: [
```

Comment:
Is this AND redundant since it only has 1 element?

Reply:
Let me verify